File: backend/src/main/java/pl/btsoftware/backend/transaction/infrastructure/api/BillItemRequest.java
package pl.btsoftware.backend.transaction.infrastructure.api;

import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.util.UUID;
import pl.btsoftware.backend.shared.CategoryId;
import pl.btsoftware.backend.transaction.application.BillItemCommand;

public record BillItemRequest(
        @NotNull UUID categoryId,
        @NotNull BigDecimal amount,
        @Nullable String description) {
    public BillItemCommand toCommand() {
        return new BillItemCommand(CategoryId.of(categoryId), amount, description);
    }
}

File: frontend/lib/models/transaction/bill_item_request.dart
class BillItemRequest {
  final String? categoryId;
  final double amount;
  final String description;

  const BillItemRequest({required this.categoryId, required this.amount, required this.description});

  Map<String, dynamic> toJson() {
    return {'categoryId': categoryId, 'amount': amount, 'description': description};
  }
}
